src/ui/ui-storyEditor.js

```diff
@@ -252,4 +293,107 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         that.events.onRemoveBlocksCompleted.fire(removedBlockKeys);
     };
 
+    /**
+     * Reorders the specified block UI component in the specified direction
+     *
+     * @param {Component} reorderer - an instance of fluid.reorderList
+     * @param {jQuery} blockUi - a jQueryable of an sjrk.storyTelling.blockUi.editor's container
+     * @param {Number} direction - a member of fluid.direction
+     */
+    sjrk.storyTelling.ui.storyEditor.reorderBlock = function (reorderer, blockUi, direction) {
+        var relativePosition = reorderer.layoutHandler.getRelativePosition(blockUi, direction);
+
+        reorderer.requestMovement(relativePosition, blockUi);
+    };
+
+    /**
+     * Updates the order of each block model according to the order of the blockUi
+     * elements in the story content editing area
+     *
+     * @param {Component} that - an instance of sjrk.storyTelling.ui.storyUi
+     */
+    sjrk.storyTelling.ui.storyEditor.updateBlockOrder = function (that) {
+        var blockUis = that.reorderer.container.children();
+
+        // used to retrieve the class name for each block
+        // within the blockManager's managedViewComponentRegistry
+        var managedClassNamePattern = sjrk.storyTelling.ui.storyEditor.getManagedClassNamePattern(that.blockManager.options.selectors.managedViewComponents);
```

Comment:
This code has changed a lot since the first review but I still think the "managedClassNamePattern" strategy is quite overcomplicated compared with the requirement - did the alternative strategies, e.g. via a "viewMapper" component or similar get written up as a JIRA somewhere?

Reply:
It has now! I've just filed [SJRK-371](https://issues.fluidproject.org/browse/SJRK-371) and have left a comment in the code pointing to it. Many thanks again for your feedback, hopefully we can get to that soon.